worker/src/lib.rs

```diff
@@ -299,7 +300,73 @@ pub async fn witness_discord(
             vc.generate_proof(
                 &sk,
                 &LinkedDataProofOptions {
-                    verification_method: Some(format!("{}#controller", SPRUCE_DIDWEB)),
+                    verification_method: Some(URI::String(format!("{}#controller", SPRUCE_DIDWEB))),
+                    ..Default::default()
+                }
+            )
+            .await
+        );
+        vc.proof = Some(OneOrMany::One(proof));
+
+        Ok(jserr!(serde_json::to_string(&vc)).into())
+    })
+}
+
+#[wasm_bindgen]
+pub async fn dns_lookup(
+    secret_key_jwk: String,
+    public_key_tezos: String,
+    domain: String,
+) -> Promise {
+    initialize_logging();
+
+    future_to_promise(async move {
+        let pk: JWK = jserr!(jwk_from_tezos_key(&public_key_tezos));
+        let sk: JWK = jserr!(serde_json::from_str(&secret_key_jwk));
+
+        let dns_result = jserr!(dns::retrieve_dns_response(domain.clone()).await);
+
+        let mut vc = jserr!(dns::build_dns_vc(&pk));
+
+        let mut signature_to_resolve = "".to_string();
+        for answer in dns_result.Answer {
+            let trimmed_signature: &str = &answer.data[1..answer.data.len() - 1];
```

Comment:
Although the extra quotation marks are widely used my understanding is that it's not a real standard, so I would do a check before removing the character at each end.

Reply:
It seems like when I use the Cloudflare API the TXT record is always returned as starting with "\" and ending with \"" so I believe trimming is necessary.

Reply:
Those quotes are in the record, it's not added by the API. What I meant what, just to be on the safe side, to check that they exist before removing them. In case there's a DNS provider out there that doesn't add them by default.